File: python/meteroid/models/add_on.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .add_on_id import AddOnId
from .entitlement import Entitlement
from .price_id import PriceId
from .product_fee_type_enum import ProductFeeTypeEnum
from .product_id import ProductId


@dataclasses.dataclass
class AddOn(BaseModel):
    created_at: datetime

    id: AddOnId

    name: str

    price_id: PriceId

    product_id: ProductId

    self_serviceable: bool

    archived_at: t.Optional[datetime] = None

    description: t.Optional[str] = None

    entitlements: t.Optional[t.List[Entitlement]] = None

    fee_type: t.Optional[ProductFeeTypeEnum] = None

    max_instances_per_subscription: t.Optional[int] = None

File: go/add_on.go
// this file is @generated
package meteroid

import "time"

type AddOn struct {
	// RFC 3339 timestamp.
	ArchivedAt *time.Time `json:"archived_at,omitempty"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	Entitlements []Entitlement `json:"entitlements,omitempty"`

	FeeType *ProductFeeTypeEnum `json:"fee_type,omitempty"`

	Id AddOnId `json:"id"`

	MaxInstancesPerSubscription *int32 `json:"max_instances_per_subscription,omitempty"`

	Name string `json:"name"`

	PriceId PriceId `json:"price_id"`

	ProductId ProductId `json:"product_id"`

	SelfServiceable bool `json:"self_serviceable"`
}
